buggy_utils.py

```
@@ -0,0 +1,107 @@
+
+
+# Unhandled None and type errors
+def calculate_average(numbers):
+    total = 0
+    for n in numbers:
+        total += n
+    return total / len(numbers)  # ZeroDivisionError if empty list
+
+
+# Mutable default argument
+def append_to_list(item, lst=[]):
```

Ruff (0.15.12)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)